### PR TITLE
Feature/Improve batch columnar validation

### PR DESCRIPTION
**Pull request by figueroa1395**
Continuation of #734: Improves batch validation for columnar data.

### CODE REVIEW

**Comment by figueroa1395** [comment]
I might have used git wrongly here. I was supposed just to use #699 as a base to do the fix, but it seems like everything has been copied. @mgovers maybe you have input on this?

**Comment by mgovers** [comment]
> I might have used git wrongly here. I was supposed just to use #699 as a base to do the fix, but it seems like everything has been copied. @mgovers maybe you have input on this?

you can try multiple things:

1. change the base branch of this PR to `feature/columnar-dataset-python-extension`
2. rebase
3. cherry-pick

File: tests/unit/validation/test_batch_validation.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def test_validate_batch_data_input_error(input_data, batch_data):
 
 
 def test_validate_batch_data_update_error(input_data, batch_data):
```
Comment (mgovers) [inline tests/unit/validation/test_batch_validation.py:82]: can you add a validation case with columnar input and columnar update dataset?

**Comment by figueroa1395** [comment]
Since #760 implemented some necessary changes for this PR and I have made a mess of myself, I will close this PR to open a new relevant one.

**Comment by figueroa1395** [comment]
Since https://github.com/PowerGridModel/power-grid-model/pull/760 implemented some necessary changes for this PR and I have made a mess of myself, I will close this PR to open a new relevant one.

**Comment by TonyXiang8787** [comment]
@figueroa1395 please delete the branch if it is not needed.

**Pull request closed without merging**